pkgs/tools/misc/fwup/default.nix

```diff
@@ -1,26 +1,49 @@
-{ stdenv, lib, fetchFromGitHub, autoreconfHook, makeWrapper, pkg-config
-, zlib, lzma, bzip2, mtools, dosfstools, zip, unzip, libconfuse, libsodium
-, libarchive, darwin, coreutils }:
+{ lib
+, autoreconfHook
+, bzip2
+, coreutils
+, darwin
+, dosfstools
+, fetchFromGitHub
+, libarchive
+, libconfuse
+, libsodium
+, lzma
+, makeWrapper
+, mtools
+, pkg-config
+, stdenv
+, unzip
+, which
+, xdelta
+, zip
+, zlib
+}:
 
 stdenv.mkDerivation rec {
   pname = "fwup";
-  version = "1.5.2";
+  version = "1.8.3";
 
   src = fetchFromGitHub {
     owner = "fhunleth";
     repo = "fwup";
     rev = "v${version}";
-    sha256 = "05sjdlh450hk474a44yr6kz9dzx72jfxpi1krxbd0pdizlmfypsg";
+    sha256 = "sha256-ayfcnIZ7MuBsCy1giwmY2D2C6AukwS+fevmXqGa4c1w=";
   };
 
   doCheck = true;
+
+  checkInputs = [ which xdelta ];
+
```

Comment:
It's more canonical to put `checkInputs` after build inputs (so here after `propagatedBuildInputs`). Yes, `doCheck` is a bit weirdly placed here, and should go after all inputs ;).